README.md
GeographicDistanceTools
================
This repo has been created for archival purposes to replace an SVN archive that is no longer online and accessible. Thiese tools are no longer being actively developed.

##What is the Geographic Distance Matrix Generator?
The Geographic Distance Matrix Generator is a platform-independent Java application that implements the same powerful suite of spherical functions as the Perpendicular Distance Calculator to compute all pair wise distances from a simple list of geographic coordinates. http://biodiversityinformatics.amnh.org/open_source/gdmg/

##What is the perpendicular distance calculator?
The Perpendicular Distance Calculator is a platform-independent Java application that implements a powerful suite of spherical functions for processing observations made during standard line transect surveys. Line transect sampling a commonly used distance sampling method that can be used for estimating the density and/or abundance of biological populations. http://biodiversityinformatics.amnh.org/open_source/pdc/

##Why is the Geographic Distance Matrix Generator useful?
In the fields of phylogeography and landscape genetics, biologists attempt to detect patterns in the distribution of genetic variation across different spatial scales. The most commonly known process is isolation by distance (IBD), under which genetic similarity decreases with geographic distance. Barriers to migration of organisms can hinder gene flow, thus increasing genetic distance, even if populations are not far apart. With the increasing availability of mobile GPS technology, every scientist can obtain latitude-longitude coordinates for target organisms, which can later be used in regression analyses in conjunction with genetic distances, or matrix associations with permutations for statistical testing, as well as multiple matrix correlations when testing for more complex biogeographic and ecological scenarios (Mantel and partial Mantel tests, respectively).The Geographic Distance Matrix Generator can be used to easily generate distance matrices needed for these types of analyses.

##Why use spherical functions?
The Geographic Distance Matrix Generator implements methods founded in spherical geometry because calculations can be performed directly with geographic coordinates, i.e., latitude and longitude, saving researchers time and money by eliminating the need for complex, expensive spatial software.


ChangeLog
2007-10-26 
	* org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculatorGui.java, 
	  org/amnh/cbc/perpendicularDistanceCalculator/About.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorGUI.java,
	  org/amnh/cbc/geographicsDistanceMatrixGenerator/About.java,
	  - Updated urls to reflect name change to Biodiversity Informatics Facility
	  
2007-03-06 13:39  pete

	* org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorGUI.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculatorGui.java,
	  resources/SplashScreenGraphic-GDMG.jpg,
	  resources/SplashScreenGraphic-PDC.jpg: 
	  -Small bug fix. Wrong URL in version checking

2007-03-02 20:57  pete

	* ChangeLog: -Added changelog

2007-03-02 14:18  pete

	* ., resources/SplashScreenGraphic-GDMG.jpg,
	  resources/SplashScreenGraphic-PDC.jpg: -Updated splash screens

2007-03-01 22:07  pete

	* org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorGUI.java:
	  -Added export feature to GDMG

2007-03-01 21:37  pete

	* org/amnh/cbc/perpendicularDistanceCalculator/About.java: -Updated
	  about dialogs for the GDMG

2007-03-01 21:15  pete

	* org/amnh/cbc/core/VersionCheck.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/About.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/About.java: -Updated
	  the version reporting class for better output display
	  -Updated about dialogs for both the PDC and GDMG

2007-03-01 15:43  pete

	* org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorEngine.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorGUI.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/LocationRecord.java:
	  -Removed StringTokenizerClass -Parse location records once on
	  load rather than each time the matrix is built (major speed
	  boost) -Added the equivalent of a LocationRecord struct to hold
	  all location information
	  GeographicDistanceMatrixGeneratorEngine.java
	  GeographicDistanceMatrixGeneratorGUI.java LocationRecord.java

2007-02-07 15:37  pete

	* org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorGUI.java:
	  Added about menu item

2007-02-07 15:32  pete

	* org/amnh/cbc/core/SimpleFileFilter.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/About.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGenerator.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorEngine.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorGUI.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/GCDWindow.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/GreatCircleDistanceGui.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/IGCPWindow.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/IntermediateGreatCirclePointsGui.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculatorGui.java:
	  Finished commenting and refactoring

2007-02-06 21:16  pete

	* org/amnh/cbc/core/SplashScreen.java,
	  org/amnh/cbc/geospatial/SphericalFunctionEngine.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/About.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/GCDWindow.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/IGCPWindow.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculator.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculatorGui.java:
	  Adding more comments to the Perpendicular Distance Calculator and
	  refactoring classes

2007-02-06 15:53  pete

	* org/amnh/cbc/core/SplashScreen.java,
	  org/amnh/cbc/core/VersionCheck.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/About.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/About.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculator.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculatorGui.java,
	  resources/SplashScreenGraphic-GDMG.jpg,
	  resources/SplashScreenGraphic-PDC.jpg, resources/cbc-blue-sm.jpg,
	  resources/nasa-sm.jpg, resources/noaa-sm.jpg: Moved a few things
	  around, but having problems with the project in Eclipse

2007-02-02 18:42  pete

	* edu, edu/stanford, edu/stanford/ejalbert,
	  edu/stanford/ejalbert/BrowserLauncher.java, org, org/amnh,
	  org/amnh/cbc, org/amnh/cbc/core,
	  org/amnh/cbc/core/SimpleFileFilter.java,
	  org/amnh/cbc/core/SplashScreen.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorEngine.java,
	  org/amnh/cbc/geographicDistanceMatrixGenerator/GeographicDistanceMatrixGeneratorGUI.java,
	  org/amnh/cbc/geospatial,
	  org/amnh/cbc/geospatial/SphericalFunctionEngine.java,
	  org/amnh/cbc/perpendicularDistanceCalculator,
	  org/amnh/cbc/perpendicularDistanceCalculator/GCDWindow.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/IGCPWindow.java,
	  org/amnh/cbc/perpendicularDistanceCalculator/PerpendicularDistanceCalculator.java,
	  resources: Initial import of base classes before refactoring

2007-02-02 18:25  pete

	* branches, tags, .: Initial repository setup



org/amnh/cbc/perpendicularDistanceCalculator/About.java
/*
** File: About.java
** Author: Peter J. Ersts (ersts@amnh.org)
** Creation Date: 2007-02-06
**
** Copyright (c) 2007, American Museum of Natural History. All rights reserved.
** 
** This library is free software; you can redistribute it and/or
** modify it under the terms of the GNU Library General Public
** License as published by the Free Software Foundation; either
** version 2 of the License, or (at your option) any later version.
** 
** This library is distributed in the hope that it will be useful,
** but WITHOUT ANY WARRANTY; without even the implied warranty of
** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
** Library General Public License for more details.
** 
** You should have received a copy of the GNU Library General Public
** License along with this library; if not, write to the
** Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, 
** MA 02110-1301, USA.
**
** This program was prepared by the the above author(s) 
** under award No. NA04AR4700191 from the National Oceanic and Atmospheric
** Administration, U.S. Department of Commerce.  The statements, findings,
** conclusions, and recommendations are those of the author(s) and do not
** necessarily reflect the views of the National Oceanic and Atmospheric
** Administration or the Department of Commerce.
**
**/
package org.amnh.cbc.perpendicularDistanceCalculator;

import java.awt.Color;
import java.awt.FlowLayout;
import java.awt.BorderLayout;

import java.io.IOException;

import javax.swing.JDialog;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.JFrame;
import javax.swing.ImageIcon;
import javax.swing.JEditorPane;
import javax.swing.BorderFactory;
import javax.swing.JScrollPane;

import javax.swing.event.HyperlinkEvent;
import javax.swing.event.HyperlinkListener;

import edu.stanford.ejalbert.BrowserLauncher;

/**
 * This class simply displays a little dialog with information about the application.
 * @author Peter J. Ersts
 *
 */
public class About extends JDialog implements HyperlinkListener {

	/**
	 * Constructor
	 * @param parent	Parent frame
	 * @param modal		Boolean to indicate if this dialog should be modal
	 */
	About(JFrame parent, String version) {
		String messageText = "<HTML><BODY>"+
	 						"Version "+ version +" of the Perpendicular Distance Calculator "+
 							"written by Peter J. Ersts, Project Specialist with the <a HREF=\"http://cbc.amnh.org\">Center for "+
 							"Biodiversity and Conservation</a> at the <a HREF=\"http://amnh.org\">American Museum of Natural History</a>. "+
 							"Eric Albert, Tim Collins, Ned Horning, Kevin Koy, Matt Leslie, and Marco Polin "+
 							"should be acknowledged for their contributions which have taken the form of code, constructive criticism, "+
 							"beta-testing and moral support. This application implements Eric Albert's BrowserLauncher class, which is know not to work on many linux systems.<BR><BR>"+
 							"Questions, comments can be posted on: <BR>"+
 							 "<a HREF=\"http://biodiversityinformatics.amnh.org/open_source/gdmg/comments.php\">http://biodiversityinformatics.amnh.org/gdmg/comments.php</a><BR><BR>"+
 							 "Bug reports and enhancements can be posted on: <a HREF=\"http://trac.biodiversityinformatics.amnh.org/projects/javageodistancetools\">http://trac.biodiversityinformatics.amnh.org/projects/javageodistancetools</a><BR><BR>"+
 							"This work has been partially supported by NASA under award No. NAG5-8543. "+
 							"Additionally, this program was prepared by the the above author(s) under "+
 							"award No. NA04AR4700191 from the National Oceanic and Atmospheric "+
 							"Administration, U.S. Department of Commerce.  The statements, findings, "+
 							"conclusions, and recommendations are those of the author(s) and do not "+
 							"necessarily reflect the views of the National Oceanic and Atmospheric "+
 							"Administration or the Department of Commerce.</BODY></HTML>";
		setSize(435,600);
		setTitle("About");
		
		getContentPane().setBackground(Color.WHITE);
		setBackground(new Color(255,255,255));
		setDefaultCloseOperation(JDialog.DISPOSE_ON_CLOSE);
		setLocation((int)parent.getLocationOnScreen().getX() + (parent.getWidth()/2) - (getWidth()/2), (int)parent.getLocationOnScreen().getY() + (parent.getHeight()/2) - (getHeight()/2));
		
		JEditorPane text = new JEditorPane();
		text.addHyperlinkListener(this);
		text.setContentType("text/html");
		text.setText(messageText);
		text.setCaretPosition(0);
		text.setEditable(false);
		JScrollPane textScrollPane = new JScrollPane(text);
		textScrollPane.setVerticalScrollBarPolicy(JScrollPane.VERTICAL_SCROLLBAR_AS_NEEDED);
		
		
		JPanel splash = new JPanel();
		splash.setBackground(new Color(255,255,255));
		splash.setBorder(BorderFactory.createEmptyBorder());
		splash.add(new JLabel(new ImageIcon(this.getClass().getClassLoader().getResource("resources/SplashScreenGraphic-PDC.jpg"))));
		getContentPane().add(splash, BorderLayout.NORTH);
		
		JPanel logos = new JPanel(new FlowLayout(FlowLayout.CENTER, 25, 0));
		logos.setBackground(new Color(255, 255, 255));
		logos.add(new JLabel(new ImageIcon(this.getClass().getClassLoader().getResource("resources/cbc-blue-sm.jpg"))));
		logos.add(new JLabel(new ImageIcon(this.getClass().getClassLoader().getResource("resources/nasa-sm.jpg"))));
		logos.add(new JLabel(new ImageIcon(this.getClass().getClassLoader().getResource("resources/noaa-sm.jpg"))));
		
		getContentPane().add(logos, BorderLayout.SOUTH);
		getContentPane().add(textScrollPane, BorderLayout.CENTER);
		setVisible(true);
	}
	
	/**
	 * Required method to implement HyperlinkListener
	 */
	public void hyperlinkUpdate(HyperlinkEvent evt) {
		if(evt.getEventType() == HyperlinkEvent.EventType.ACTIVATED) {
			try {
				BrowserLauncher.openURL(evt.getURL().toString());
			}
			catch (IOException e) {}
		}
	}
}
